### PR TITLE
Fix oneof with reserved keywords as names

### PR DESCRIPTION
**Pull request by pranjalvachaspati-toast**
This fixes issue #154

### CODE REVIEW

File: protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/OneofAnnotator.kt [hunk 1]
```diff
@@ -77,13 +77,13 @@ private constructor(
                                 }
                             }
                             .addProperty(
-                                PropertySpec.builder(v.fieldName, v.type)
-                                    .initializer(v.fieldName)
+                                PropertySpec.builder(v.fieldName.removePrefix("`").removeSuffix("`"), v.type)
```
Comment (andrewparmet) [inline protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/OneofAnnotator.kt:80]: I think this is done somewhere else too - is it worth extracting this into a utility?

Separately - I'm not sure why we have to do this. Do you know why things get escaped twice?

File: testing/runtime-tests/src/main/proto/toasttab/protokt/testing/rt/oneof/reserved_name.proto [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package toasttab.protokt.testing.rt.oneof;
+
+message Example {
+    oneof my_oneof {
```
Comment (andrewparmet) [inline testing/runtime-tests/src/main/proto/toasttab/protokt/testing/rt/oneof/reserved_name.proto:21]: nit: two space indents in proto.

File: protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/OneofAnnotator.kt (hunk 1 above)
Comment (pranjalv123) [reply protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/OneofAnnotator.kt:80]: I think it's a kotlinpoet bug - I'm going to look into it, and hopefully then we can get rid of this workaround.

Comment (pranjalvachaspati-toast) [reply protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/OneofAnnotator.kt:80]: I think it's a kotlinpoet bug - I'm going to look into it, and hopefully then we can get rid of this workaround.

File: testing/runtime-tests/src/main/proto/toasttab/protokt/testing/rt/oneof/reserved_name.proto (hunk 2 above)
Comment (pranjalvachaspati-toast) [reply testing/runtime-tests/src/main/proto/toasttab/protokt/testing/rt/oneof/reserved_name.proto]: fixed

**Comment by pranjalvachaspati-toast** [comment]
Actually it looks like there's a fix for this in a newer version of kotlinpoet already. I'll make a new PR for that.

File: protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/OneofAnnotator.kt (hunk 1 above)
Comment (andrewparmet) [reply protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/OneofAnnotator.kt:80]: Other instances:

https://github.com/open-toast/protokt/blob/main/protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/MessageAnnotator.kt#L102

https://github.com/open-toast/protokt/blob/main/protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/MessageAnnotator.kt#L128

https://github.com/open-toast/protokt/blob/main/protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DslAnnotator.kt#L68

**Pull request closed without merging**